src/utilities.ts

```diff
@@ -12,13 +16,22 @@ export function calculateTotalRegistrationFee(
   fees: Fees,
   tier: ServiceTier,
   years: number,
+  currentTimestamp: number, // block timestamp
 ) {
   // Initial cost to register a name
   const initialNamePurchaseFee = fees[name.length.toString()];
 
   // total cost to purchase name and tier
   return (
-    initialNamePurchaseFee + calculateAnnualRenewalFee(name, fees, tier, years)
+    initialNamePurchaseFee +
+    calculateAnnualRenewalFee(
+      name,
+      fees,
+      tier,
+      years,
+      DEFAULT_UNDERNAME_COUNT,
```

Comment:
i forget where we left this:
 - would renenwing a lease would reset undernames count?
 
i think `undernameCount` should be passed into `calculateTotalRegistrationFee` here

Reply:
undername count is only reset if a lease is expired, if the lease is being extended the cost of the lease includes the cost of the undernames.

Reply:
Far as I can tell everywhere `calculateTotalRegistrationFee` is used, its to calculate the fee for an expired name, which when registered would be registered with the default undername count (because undernames expire the same time as the arns name).

In extend record action, we have calculateAnnualRenewal which does take in undernames though.

Does that seem right?